eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/eventdetails/BasicEventDetails.java

```diff
@@ -55,54 +59,43 @@ public BasicEventDetails(@NonNull Event event)
-     */
-    private final static class StartEventDetailsRunnable implements Runnable
-    {
-        private final Activity mActivity;
-        private final Event mEvent;
-
+        final OptionalArgument<MicroFragmentHost> host = new OptionalArgument<>(Keys.MICRO_FRAGMENT_HOST, activity);
 
-        private StartEventDetailsRunnable(@NonNull Activity activity, @NonNull Event event)
+        if (!host.isPresent())
         {
-            mActivity = activity;
-            mEvent = event;
+            throw new UnsupportedOperationException("Opening EventDetails with already loaded Event, from non-host Activity is not implemented.");
         }
 
-
-        @Override
-        public void run()
+        new LoadActionsFromCacheTask(mEvent, new SafeAsyncTaskCallback<Event, Optional<List<Link>>>()
         {
-            Intent intent = new Intent(mActivity, MicroFragmentHostActivity.class);
-            Bundle nestedBundle = new Bundle();
-
-            FutureServiceConnection<ActionService> actionService = new ActionService.FutureConnection(mActivity);
-            try
-            {
-                List<Link> actions = actionService.service(40).cachedActions(mEvent.uid());
-                // Start the details with the actions from the cache.
-                nestedBundle.putParcelable("MicroFragment", new ShowEventMicroFragment(mEvent, actions));
-            }
-            catch (InterruptedException | TimeoutException | NoSuchElementException e)
+            @Override
+            public void onTaskFinish(SafeAsyncTaskResult<Optional<List<Link>>> result, Event event)
             {
-                // An error occurred or the actions are not in the cache yet, let the details activity load the actions.
-                nestedBundle.putParcelable("MicroFragment", new ActionLoaderMicroFragment(mEvent));
+                Optional<List<Link>> actions = safeResult(result);
+
+                MicroFragment microFragment = actions.isPresent() ?
+                        new ShowEventMicroFragment(mEvent, actions.value()) : new ActionLoaderMicroFragment(mEvent);
+
+                // TODO This couldn't be called from background thread because ForwardTransition's TimeStamp has to be created on main thread.
+                // Should we create TimeStamp beforehand to be able to fire the transtion without calling back to main?
+                host.value().execute(activity, new Swiped(new ForwardTransition<>(microFragment)));
             }
-            finally
+
+
+            private Optional<List<Link>> safeResult(SafeAsyncTaskResult<Optional<List<Link>>> result)
             {
-                if (actionService.isConnected())
+                try
                 {
-                    actionService.disconnect();
+                    return result.value();
+                }
+                catch (Exception e)
+                {
+                    // Should not happen
+                    throw new RuntimeException(e);
```

Comment:
Yes, you can fire transactions from a background thread. The ParcelableDovecote (which is used internally by the MF framework) makes sure that the pigeon is always delivered on the main thread: https://github.com/dmfs/PigeonPost/blob/master/localbroadcast-pigeon/src/main/java/org/dmfs/pigeonpost/localbroadcast/ParcelableDovecote.java#L97